upup/pkg/fi/cloudup/urls.go

```diff
@@ -17,74 +17,162 @@ limitations under the License.
 package cloudup
 
 import (
+	"net/url"
 	"os"
 	"strings"
 
+	"path"
+
 	"github.com/golang/glog"
 	"k8s.io/kops"
+	"k8s.io/kops/pkg/assets"
+	"k8s.io/kops/util/pkg/hashing"
 )
 
-// baseUrl caches the BaseUrl value
-var baseUrl string
+var kopsUrl = "https://kubeupv2.s3.amazonaws.com/kops/" + kops.Version + "/"
+var baseUrl *url.URL
 
 // BaseUrl returns the base url for the distribution of kops - in particular for nodeup & docker images
-func BaseUrl() string {
-	if baseUrl != "" {
+func BaseUrl() (*url.URL, error) {
+	if baseUrl != nil {
 		// Avoid repeated logging
-		return baseUrl
+		return baseUrl, nil
 	}
 
-	baseUrl = os.Getenv("KOPS_BASE_URL")
-	if baseUrl == "" {
-		baseUrl = "https://kubeupv2.s3.amazonaws.com/kops/" + kops.Version + "/"
-		glog.V(4).Infof("Using default base url: %q", baseUrl)
+	baseUrlString := os.Getenv("KOPS_BASE_URL")
+	if baseUrlString == "" {
+		baseUrlString = kopsUrl
+		glog.V(4).Infof("Using default base url: %q", kopsUrl)
 	} else {
-		glog.Warningf("Using base url from KOPS_BASE_URL env var: %q", baseUrl)
+		// TODO I do not think we need to do this with
```

Comment:
I think that's the point of your URL code, right?  So yes, doesn't make sense to still have it.

Reply:
Ping on this

Reply:
> So yes, doesn't make sense to still have it.

I removed the code the removed the `/`.  If that is what you are referring to.